src/browser/BrowserOptionDialog.cpp

```diff
@@ -150,3 +174,48 @@ void BrowserOptionDialog::showProxyLocationFileDialog()
                                                       fileTypeFilter);
     m_ui->customProxyLocation->setText(proxyLocation);
 }
+
+void BrowserOptionDialog::removeSelectedKey()
+{
+    if (QMessageBox::Yes != MessageBox::question(this,
+            tr("Delete the selected key?"),
+            tr("Do you really want to delete the selected key?\n"
+               "This may cause the affected plugins to malfunction."),
```

Comment:
What does it mean to malfunction? I think the correct verbiage here is "This may prevent connection to the browser plugin."

Note that we warn when you remove an individual key, but when you click on "Disconnect all browsers" it just does it without any confirmation. Same with "Forget all remembered permissions" which is a little bit confusing, this button should probably say "Forget all site-specific settings on entries".

Reply:
I agree. There should be confirmation for each of these actions.